src/Microsoft.SymbolStore.Client/WindowsSymbolSever.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.SymbolStore.Client
+{
+    public class SymbolServerResult
+    {
+        public Stream Stream { get; set; }
+        public long ContentLength { get; set; }
+        public bool Compressed { get; set; }
+
+        public SymbolServerResult(Stream result, long contentLength, bool compressed)
+        {
+            Stream = result;
+            ContentLength = contentLength;
+            Compressed = compressed;
+        }
+        public SymbolServerResult(WebResponse response, bool compressed)
+        {
+            Stream = response.GetResponseStream();
+            ContentLength = response.ContentLength;
+            Compressed = compressed;
+        }
+
+        public SymbolServerResult(string fileName)
+        {
+            Stream = File.OpenRead(fileName);
+            ContentLength = new FileInfo(fileName).Length;
+            Compressed = false;
+        }
+    }
+
+    public interface ISymbolServer
+    {
+        bool PreferThisServer { get; set; }
+
+        bool IsRemoteServer { get; }
+        bool IsReachable();
+        bool IsReachable(int timeout);
+        
+        SymbolServerResult FindPEFile(string filename, int buildTimeStamp, int imageSize);
+        Task<SymbolServerResult> FindPEFileAsync(string filename, int buildTimeStamp, int imageSize);
+        SymbolServerResult FindPdb(string pdbName, Guid guid, int age);
+        Task<SymbolServerResult> FindPdbAsync(string pdbName, Guid guid, int age);
+    }
+
+    public class WindowsSymbolSever : ISymbolServer
```

Comment:
I'd like to propose doing some refactoring, but probably easier to chat in person. I have some different ideas to bounce off you and we can come to some agreement about what happens pre-checkin vs. later. Thanks for putting this all together Lee!